pkg/plugin/cb/middleware.go

```diff
@@ -0,0 +1,66 @@
+package cb
+
+import (
+	"net/http"
+
+	"github.com/Knetic/govaluate"
+	"github.com/afex/hystrix-go/hystrix"
+	"github.com/felixge/httpsnoop"
+	"github.com/hellofresh/janus/pkg/api"
+	janusErr "github.com/hellofresh/janus/pkg/errors"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	defaultPredicate = "statusCode == 0 || statusCode >= 500"
+)
+
+// NewCBMiddleware creates a new cb middleware
+func NewCBMiddleware(cfg Config, def *api.Definition) func(http.Handler) http.Handler {
+	return func(handler http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			logger := log.WithFields(log.Fields{
+				"name":                    def.Name,
+				"timeout":                 cfg.Timeout,
+				"max_concurrent_requests": cfg.MaxConcurrentRequests,
+				"error_percent_threshold": cfg.ErrorPercentThreshold,
+			})
+
+			logger.Debug("Starting cb middleware")
+			if cfg.Predicate == "" {
+				cfg.Predicate = defaultPredicate
+			}
+
+			expression, err := govaluate.NewEvaluableExpression(cfg.Predicate)
+			if err != nil {
+				log.WithError(err).Error("could not create an expression with this predicate")
+				handler.ServeHTTP(w, r)
+				return
+			}
+
+			err = hystrix.Do(def.Name, func() error {
```

Comment:
quick question, haven't we decided to make use a map in order to avoid the issues we've had with the old implementation?

Reply:
I mean have a map of the proxy name + the backend target:

- `proxy1-name:backend-service000.company.io`
- `proxy1-name:backend-service001.company.io`

Allowing us to have one circuit breaker for each target, although I don't remember when we discussed about it would be possible to do it in the middleware layer